pkg/sentry/socket/hostinet/socket_unsafe.go

```diff
@@ -55,8 +56,116 @@ func writev(fd int, srcs []unix.Iovec) (uint64, error) {
 	return uint64(n), nil
 }
 
+// interfaceIoctl implements interface requests.
+func interfaceIoctl(ctx context.Context, _ usermem.IO, arg uintptr, ifr *linux.IFReq) error {
+	var (
+		iface inet.Interface
+		index int32
+		found bool
+	)
+
+	// Find the relevant device.
+	stk := inet.StackFromContext(ctx)
+	if stk == nil {
+		return linuxerr.ENODEV
+	}
+
+	// Find the relevant device.
+	for index, iface = range stk.Interfaces() {
+		if iface.Name == ifr.Name() {
+			found = true
+			break
+		}
+	}
+	if !found {
+		return linuxerr.ENODEV
+	}
+
+	switch arg {
+	case linux.SIOCGIFINDEX:
+		// Copy out the index to the data.
+		hostarch.ByteOrder.PutUint32(ifr.Data[:], uint32(index))
+
+	case linux.SIOCGIFADDR:
+		// Copy the IPv4 address out.
+		for _, addr := range stk.InterfaceAddrs()[index] {
+			// This ioctl is only compatible with AF_INET addresses.
+			if addr.Family != linux.AF_INET {
+				continue
+			}
+			copy(ifr.Data[4:8], addr.Addr)
+			break
+		}
+
+	case linux.SIOCGIFHWADDR:
+		// Copy the hardware address out.
+		//
+		// Refer: https://linux.die.net/man/7/netdevice
+		// SIOCGIFHWADDR, SIOCSIFHWADDR
+		//
+		// Get or set the hardware address of a device using
+		// ifr_hwaddr. The hardware address is specified in a struct
+		// sockaddr. sa_family contains the ARPHRD_* device type,
+		// sa_data the L2 hardware address starting from byte 0. Setting
+		// the hardware address is a privileged operation.
+		hostarch.ByteOrder.PutUint16(ifr.Data[:], iface.DeviceType)
+		n := copy(ifr.Data[2:], iface.Addr)
+		for i := 2 + n; i < len(ifr.Data); i++ {
+			ifr.Data[i] = 0 // Clear padding.
```

Comment:
Is this necessary? `ifr.Data` should already be zeroed.

Reply:
This code comes from https://github.com/google/gvisor/blob/master/pkg/sentry/socket/netstack/netstack.go#L3353, actually ifr.Data is CopyIn args[2] from user, maybe clear the data here is better?

Reply:
Gotcha, you're correct to clear it here.